Packs/HYASInsight/Integrations/HYASInsight/HYASInsight.py

```diff
@@ -0,0 +1,838 @@
+    if raw_api_response:
+        flatten_json_response = get_flatten_json_response(raw_api_response)
+
+    return CommandResults(
+        readable_output=whois_historic_lookup_to_markdown(flatten_json_response, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{WHOIS_SUB_CONTEXT}',
+        outputs_key_field='',
+        outputs=raw_api_response,
+        # outputs=[whois_historic_build_result_context(r) for r in new_response],
+    )
+
+
+@logger
+def get_whois_current_records_by_domain(client, args):
+    whois_current_record: List[Any] = []
+    indicator_type = DOMAIN_PARAM
+    indicator_value = args.get('domain')
+    check_valid_indicator_value(indicator_type, indicator_value)
+    title = get_command_title_string(WHOIS_CURRENT_SUB_CONTEXT, indicator_type, indicator_value)
+    end_point = WHOIS_ENDPOINT
+
+    api_response = client.fetch_data_from_hyas_api(end_point, indicator_type, indicator_value, True, 'POST')
+    if api_response:
+        whois_current_record = api_response["items"] if api_response["items"] else []
+
+    return CommandResults(
+        readable_output=whois_current_lookup_to_markdown(whois_current_record, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{WHOIS_CURRENT_SUB_CONTEXT}',
+        outputs_key_field='',
+        outputs=api_response,
+        # outputs=[whois_current_build_result_context(r) for r in new_response],
+    )
+
+
+@logger
+def get_malware_samples_records_by_indicator(client, args):
+    indicator_type = args.get('indicator_type')
+    indicator_value = args.get('indicator_value')
+
+    check_valid_indicator_type(indicator_type, MALWARE_QUERY_PARAMS)
+    check_valid_indicator_value(indicator_type, indicator_value)
+    title = get_command_title_string(MALWARE_SUB_CONTEXT, indicator_type, indicator_value)
+    end_point = MALWARE_ENDPOINT
+
+    api_response = client.fetch_data_from_hyas_api(end_point, indicator_type, indicator_value, False, 'POST')
+
+    return CommandResults(
+        readable_output=malware_samples_lookup_to_markdown(api_response, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{MALWARE_SUB_CONTEXT}',
+        outputs_key_field='',
+        outputs=api_response,
+        # outputs=[malware_samples_build_result_context(r) for r in new_response],
+    )
+
+
+@logger
+def get_associated_ips_by_hash(client, args):
+    indicator_type = MD5_PARAM
+    indicator_value = args.get('md5')
+
+    check_valid_indicator_value(indicator_type, indicator_value)
+    title = get_command_title_string(HASH_IP_SUB_CONTEXT, indicator_type, indicator_value)
+    end_point = MALWARE_ENDPOINT
+
+    associated_ips = []
+    api_response = client.fetch_data_from_hyas_api(end_point, indicator_type, indicator_value, False, 'POST')
+    if api_response:
+        for obj in api_response:
+            if obj['ipv4']:
+                associated_ips.append({'ip': str(obj['ipv4'])})
+
```

Comment:
Adding to output the given MD5 in order to create association between IPs and md5 in outputs.
Please make sure to change outputs for this command in YML
```suggestion
    associated_ips = [str(obj['ipv4']) for obj in api_response if 'ipv4' in obj]
    outputs = {'md5': indicator_value, 'ips': associated_ips}

```